altapay.php

```diff
@@ -0,0 +1,2484 @@
+                ]
+            );
+            die();
+        } // RELEASE
+        elseif ($action == 'Release') {
+            try {
+                $api->release($paymentID, $action);
+                updatePaymentStatus($paymentID, 'Payment Released');
+            } catch (Exception $e) {
+                $response = json_decode($e->getMessage(), true);
+                saveLastErrorMessage($paymentID, $response['responseMsg']);
+
+                echo json_encode(
+                    [
+                        'status'  => $response['responseResult'],
+                        'message' => 'Could not release reservation. ' . $response['responseResult'] . ': '
+                                     . $response['responseMsg']
+                    ]
+                );
+                die();
+            }
+            echo json_encode(
+                [
+                    'status'  => 'success',
+                    'message' => 'Reservation released successfully'
+                ]
+            );
+            die();
+        }
+    }
+
+    /* ******************************** */
+
+    /**
+     * Method for generating order lines from order backend
+     *
+     * @param array      $orderLines
+     * @param array|null $orderLineGiftWrap
+     * @param string     $orderID
+     * @param bool       $goodWillRefund
+     *
+     * @return array
+     */
+    private function populateOrderLinesFromPost(
+        $orderLines,
+        $orderLineGiftWrap = null,
+        $orderID,
+        $goodWillRefund = false
+    ) {
+        $i                             = 0;
+        $priceAfterDiscountRounded     = 0;
+        $priceAfterDiscount            = 0;
+        $totalQuantity                 = 0;
+        $compensationAmountPerQuantity = 0;
+        $altapayOrderLines             = [];
+        $discountPercentage            = 0;
+
+
+        $orderDetail         = new Order((int)$orderID);
+        $productDetailObject = new OrderDetail;
+
+        $productDetail = $productDetailObject->getList($orderID);
+
+        $compensationQuantity = 0;
+
+        foreach ($orderLines as $key => $orderedQuantity) {
+            if ($orderedQuantity > 0) {
+                $productDetails = $productDetail[$key];
+                if ($productDetails) {
+                    $productName = $productDetails['product_name'];
+
+                    $priceWithoutReductionTaxIncl = $productDetails['unit_price_tax_incl'] / (1 - ($productDetails['reduction_percent'] / 100));
+                    $basePrice = $productDetails['original_product_price'];
+
+
+                    $cartRuleDiscounts = $this->getCartRuleDiscounts($orderDetail);
+                    // Calculation of base price
+
+                    if ($productDetails['reduction_percent'] > 0) {
+                        $discountPercentage = $productDetails['reduction_percent'];
+                    } elseif (!empty($cartRuleDiscounts)) {
```

Comment:
Since we know that $cartRuleDiscounts has been initialized on the previous line there is no need for using empty(), which could leave to hiding a typo in the code.